core/templates/dev/head/domain/exploration/StatesObjectFactory.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of State
+ * domain objects given a list of backend state dictionaries.
+ */
+
+ oppia.factory('StatesObjectFactory', ['StateObjectFactory',
+  function(StateObjectFactory) {
+    var create = function(statesDict) {
```

Comment:
Could you make this follow the pattern of a "standard object factory" a bit more? I.e. define a var States etc. following existing patterns. Later we might add something like states.getState(stateName) to encapsulate better, but in the meantime let's try and keep everything consistent.